### PR TITLE
MOB-2980

### PR DESCRIPTION
**Pull request by nurito-burrito**
Coverage endpoint added

### CODE REVIEW

File: KarhooSDK/Service/Quote/Interactor/CoverageInteractor.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  CoverageInteractor.swift
+//  KarhooSDK
+//
+//  Created by Nurseda Balcioglu on 10/11/2020.
+//  Copyright © 2020 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+protocol CoverageInteractor: KarhooExecutable {
```
Comment (EdwardW2) [inline KarhooSDK/Service/Quote/Interactor/CoverageInteractor.swift:11]: Now that I think about it, maybe rename all instances of Coverage to QuoteCoverage instead. Coverage is very vague on its own so may want to be specific with the name and call it QuoteCoverage.

**Review by EdwardW2 — CHANGES_REQUESTED** [review]
Rename coverage to QuoteCoverage

File: KarhooSDK/Api/Response/Coverage.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//  Coverage.swift
+//  KarhooSDK
+//
+//  Created by Nurseda Balcioglu on 10/11/2020.
+//  Copyright © 2020 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+public struct Coverage : KarhooCodableModel {
```
Comment (JeevanThandi) [inline KarhooSDK/Api/Response/Coverage.swift:11]:
```suggestion
public struct Coverage: KarhooCodableModel {
```

File: KarhooSDK/Service/Quote/Interactor/KarhooCoverageInteractor.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    init(requestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared)) {
+        self.coverageRequestSender = requestSender
+    }
+    
+    func set(coverageRequest: CoverageRequest) {
+        self.coverageRequest = coverageRequest
+    }
+    
+    func execute<T: KarhooCodableModel>(callback: @escaping CallbackClosure<T>) {
+        coverageRequestSender.requestAndDecode(payload: nil, endpoint: .coverage, callback: callback)
```
Comment (JeevanThandi) [inline KarhooSDK/Service/Quote/Interactor/KarhooCoverageInteractor.swift:24]: payload is still nil

File: KarhooSDKTests/Mocks/Api/Response/CoverageMock.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        return coverage
+    }
+
+    func set(coverage: Bool) -> CoverageMock {
+        create(coverage: coverage)
+        return self
+    }
+
+    private func create(coverage: Bool = false) {
+
```
Comment (JeevanThandi) [inline KarhooSDKTests/Mocks/Api/Response/CoverageMock.swift:30]: spaces

File: KarhooSDKTests/TestCases/Service/Quote/KarhooCoverageSpec.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+//  Created by Nurseda Balcioglu on 10/11/2020.
+//  Copyright © 2020 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+@testable import KarhooSDK
+
+final class KarhooCoverageSpec: XCTestCase {
```
Comment (JeevanThandi) [inline KarhooSDKTests/TestCases/Service/Quote/KarhooCoverageSpec.swift:14]:
```suggestion
final class KarhooCoverageInteractorSpec: XCTestCase {
```

File: KarhooSDKTests/TestCases/Service/Quote/KarhooCoverageSpec.swift [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    private var testobject: KarhooQuoteService!
+    private var mockCoverageInteractor = MockCoverageInteractor()
```
Comment (JeevanThandi) [inline KarhooSDKTests/TestCases/Service/Quote/KarhooCoverageSpec.swift:17]: the mock coverage interactor is for QuoteServiceSpec. (missing test) For this suite of tests you need to mock the request sender.  The testObject should be the KarhooCoverageInteractor, not the quote service. That's in the QuoteServiceSpec class you want to do that.

File: KarhooSDK/Service/Quote/Interactor/CoverageInteractor.swift (hunk 1 above)
Comment (jsm20) [reply KarhooSDK/Service/Quote/Interactor/CoverageInteractor.swift:11]: @jeremyauclert if we're planning on keeping naming in line on both platforms then we probably need to do this for Android as well.

**Review by JeevanThandi — CHANGES_REQUESTED** [review]
KarhooQuoteServiceSpec is missing success / fail tests. Interactor tests are needed that do not mock the interactor, but test it.  Checkout other interactor tests to see how RequestSender is mocked 👍

**Comment by EdwardW2** [comment]
> Rename coverage to QuoteCoverage

Might be a good idea to rename the files as well so they match what is inside them.

**Review by EdwardW2 — DISMISSED** [review]

File: KarhooSDK/Api/Request/QuoteCoverageRequest.swift [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+//
+//  Created by Nurseda Balcioglu on 10/11/2020.
+//  Copyright © 2020 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+public struct QuoteCoverageRequest: KarhooCodableModel {
+
+   public let latitude: String
```
Comment (JeevanThandi) [inline KarhooSDK/Api/Request/QuoteCoverageRequest.swift:13]: missing public init- otherwise this cant be created from outside the SDK

**Review by JeevanThandi — CHANGES_REQUESTED** [review]
public init missing for QuoteCoverageRequest - then its all good 👍

**Review by JeevanThandi — APPROVED** [review]

**Pull request merged**